test/integration/scheduler/queue_test.go

```diff
@@ -247,14 +250,88 @@ func TestCoreResourceEnqueue(t *testing.T) {
+					return fmt.Errorf("failed to update the pod: %w", err)
+				}
+				return nil
+			},
+			wantRequeuedPods:          sets.New("pod1"),
+			enableSchedulingQueueHint: []bool{false, true},
+		},
+		{
+			name:        "Pod got resource scaled down requeued to activeQ",
+			initialNode: st.MakeNode().Name("fake-node").Capacity(map[v1.ResourceName]string{v1.ResourceCPU: "2"}).Obj(),
+			pods: []*v1.Pod{
+				// - Pod1 requests a large amount of CPU and will be rejected by the NodeResourcesFit plugin.
+				st.MakePod().Name("pod1").Req(map[v1.ResourceName]string{v1.ResourceCPU: "4"}).Container("image").Obj(),
+			},
+			triggerFn: func(testCtx *testutils.TestContext) error {
+				// Trigger a PodUpdate event by reducing cpu requested by pod1.
+				// It makes Pod1 schedulable.
+				if _, err := testCtx.ClientSet.CoreV1().Pods(testCtx.NS.Name).Update(testCtx.Ctx, st.MakePod().Name("pod1").Req(map[v1.ResourceName]string{v1.ResourceCPU: "2"}).Container("image").Obj(), metav1.UpdateOptions{}); err != nil {
+					return fmt.Errorf("failed to update the pod: %w", err)
+				}
+				return nil
+			},
+			wantRequeuedPods:          sets.New("pod1"),
+			enableSchedulingQueueHint: []bool{false, true},
+		},
+		{
+			name:        "Updating pod's condition doesn't make it requeued to activeQ",
+			initialNode: st.MakeNode().Name("fake-node").Taints([]v1.Taint{{Key: v1.TaintNodeNotReady, Effect: v1.TaintEffectNoSchedule}}).Obj(),
+			pods: []*v1.Pod{
+				// - Pod1 doesn't have the required toleration and will be rejected by the TaintToleration plugin.
+				st.MakePod().Name("pod1").Container("image").Obj(),
+			},
+			// Simulate a Pod update by directly calling `SchedulingQueue.Update` instead of actually updating a Pod
+			// because we don't have a way to confirm the scheduler has handled a Pod update event at the moment.
+			// TODO: actually update a Pod update and confirm the scheduler has handled a Pod update event with a metric.
+			// https://github.com/kubernetes/kubernetes/pull/122234#discussion_r1597456808
+			triggerFn: func(testCtx *testutils.TestContext) (err error) {
+				// Trigger a Pod Condition update event.
+				// It will not make pod1 schedulable
+				var (
+					oldPod *v1.Pod
+					newPod *v1.Pod
+				)
+				if oldPod, err = testCtx.ClientSet.CoreV1().Pods(testCtx.NS.Name).Get(testCtx.Ctx, "pod1", metav1.GetOptions{}); err != nil {
+					return fmt.Errorf("failed to get the pod: %w", err)
+				}
+				newPod = oldPod.DeepCopy()
+				newPod.Status.Conditions[0].Message = "injected message"
+
+				if err := testCtx.Scheduler.SchedulingQueue.Update(
+					klog.FromContext(testCtx.Ctx),
+					oldPod,
+					newPod,
+				); err != nil {
+					return fmt.Errorf("failed to update the pod: %w", err)
+				}
+				return nil
+			},
+			wantRequeuedPods:          sets.Set[string]{},
+			enableSchedulingQueueHint: []bool{true},
```

Comment:
```suggestion
			// This behaviour is only true when enabling QHint 
			// because QHint of TaintToleration would decide to ignore a Pod update.
			enableSchedulingQueueHint: []bool{true},
```